File: src/java/DeleteNode.java
package org.jy;


class ListNode {
    int val;
    ListNode next;

    ListNode(int x) {
        val = x;
    }
}

// https://leetcode-cn.com/problems/delete-node-in-a-linked-list/solution/
public class DeleteNode {

    public void deleteNode(ListNode node) {
        node.val = node.next.val;
        node.next = node.next.next;
    }

}

File: src/go/DeleteNode.go
package main

type org.jy.ListNode struct {
	Val  int
	Next *org.jy.ListNode
}

// https://leetcode-cn.com/problems/delete-node-in-a-linked-list/solution/
func deleteNode(node *org.jy.ListNode) {
	node.Val = node.Next.Val
	node.Next = node.Next.Next
}
